src/tailwindcss.worker.ts

```diff
@@ -103,13 +112,16 @@ async function stateFromConfig(
       },
       // eslint-disable-next-line require-await
       async getConfiguration() {
+        const { editor, tailwindCSS } = intellisense || {}
+        const { classAttributes = [], lint = {}, ...tailwindOptions } = tailwindCSS || {}
+
         return {
-          editor: { tabSize: 2 },
+          editor: { tabSize: 2, ...editor },
           // Default values are based on
           // https://github.com/tailwindlabs/tailwindcss-intellisense/blob/v0.9.1/packages/tailwindcss-language-server/src/server.ts#L259-L287
           tailwindCSS: {
             emmetCompletions: false,
-            classAttributes: ['class', 'className', 'ngClass'],
+            classAttributes: ['class', 'className', 'ngClass', ...classAttributes],
```

Comment:
Should the user provided class attributes extend the defaults? Or should it replace them? What does the language server do?

Reply:
I know that in VS Code you can directly edit the list of attributes within the settings panel or json. Here, we have to go another route. We either extend or override. In my opinion, we should extend with at least `class` and `className` as default. You wouldn't want to pass `['styles']` and lose completion/hover on `class` and `className`. I think the expected behavior is that `styles` gets added to the existing list.

If we really want to leave the choice to the user:

``` typescript
// classAttributes would look like this
type classAttributes = { extend: string[] } | { override: string[] } 

// and the implementation like this
tailwindCSS: {
  classAttributes: 'extend' in classAttributes
    ? ['class', 'className', ...classAttributes.extend]
    : classAttributes.override
}
```